### ISSUE TITLE
400 bad request when trying to login after creating admin user

### ISSUE THREAD

**Issue by Jomack16** [aunefyren/wrapperr#72]
After creating the admin username and password, trying to login to the admin page gives the following error:
400 Bad Request
This is happening for this call:
POST http://192.168.1.61:8282/api/get/wrapperr-version

This is my create statement for the Docker container:
docker create \
  --name=plexwrapper \
  --hostname=plexwrapper \
  -e TZ=America/Chicago \
  -p 8282:8282 \
  --restart=unless-stopped \
  aunefyren/wrapperr

I have tried creating a volume for the config as well (-v /opt/DockerAppdata/plexwrapper:/app/config). Same issue.

**Comment by aunefyren** [comment]
Are you able to get the logs, either from the Docker or Wrapperr?

```
$ docker logs plexwrapper 
```

or wrapperr.log in app/config/

**Comment by Jomack16** [comment]
[wrapperr.log](https://github.com/aunefyren/wrapperr/files/13284577/wrapperr.log)

Yes, uploaded the wrapperr.log here.

**Comment by aunefyren** [comment]
Huh, very odd. If you utilize:

```
$ docker logs plexwrapper 
```

can you see which API endpoint is returning a 400? Is there no error message returned in your browser?

If you use `inspect` within your browser, could you also look at the network calls? Perhaps see the endpoint and returned body there?

**Comment by Jomack16** [comment]
___       __    ________   ________   ________   ________   _______    ________   ________
|\  \     |\  \ |\   __  \ |\   __  \ |\   __  \ |\   __  \ |\  ___ \  |\   __  \ |\   __  \
\ \  \    \ \  \\ \  \|\  \\ \  \|\  \\ \  \|\  \\ \  \|\  \\ \   __/| \ \  \|\  \\ \  \|\  \
 \ \  \  __\ \  \\ \   _  _\\ \   __  \\ \   ____\\ \   ____\\ \  \_|/__\ \   _  _\\ \   _  _\
  \ \  \|\__\_\  \\ \  \\  \|\ \  \ \  \\ \  \___| \ \  \___| \ \  \_|\ \\ \  \\  \|\ \  \\  \|
   \ \____________\\ \__\\ _\ \ \__\ \__\\ \__\     \ \__\     \ \_______\\ \__\\ _\ \ \__\\ _\
    \|____________| \|__|\|__| \|__|\|__| \|__|      \|__|      \|_______| \|__|\|__| \|__|\|__|
 ____________   ____________   ____________   ____________   ____________   ____________   ____________
|\____________\|\____________\|\____________\|\____________\|\____________\|\____________\|\____________\
\|____________|\|____________|\|____________|\|____________|\|____________|\|____________|\|____________|

Log file set.
2023/11/08 04:31:00 Running version v3.2.0.
2023/11/08 04:31:00 Starting Wrapperr on port 8282.
2023/11/08 04:31:00 Starting using HTTP.
[GIN] 2023/11/08 - 12:44:34 | 200 |      2.6434ms |   192.168.1.219 | GET      "/"
[GIN] 2023/11/08 - 12:44:34 | 200 |    8.451228ms |   192.168.1.219 | GET      "/js/functions.js"
[GIN] 2023/11/08 - 12:44:34 | 200 |    19.24283ms |   192.168.1.219 | GET      "/assets/css/wrapped.css"
[GIN] 2023/11/08 - 12:44:34 | 200 |     474.235µs |   192.168.1.219 | GET      "/js/index.js"
[GIN] 2023/11/08 - 12:44:34 | 200 |     188.108µs |   192.168.1.219 | GET      "/js/get_functions.js"
[GIN] 2023/11/08 - 12:44:34 | 200 |     493.034µs |   192.168.1.219 | GET      "/js/get_stats.js"
[GIN] 2023/11/08 - 12:44:34 | 200 |     363.581µs |   192.168.1.219 | GET      "/assets/img/gift.svg"
[GIN] 2023/11/08 - 12:44:34 | 200 |     928.396µs |   192.168.1.219 | GET      "/assets/loading.gif"
[GIN] 2023/11/08 - 12:44:34 | 200 |     252.263µs |   192.168.1.219 | GET      "/assets/external-link.svg"
[GIN] 2023/11/08 - 12:44:34 | 200 |     233.971µs |   192.168.1.219 | GET      "/assets/done.svg"
[GIN] 2023/11/08 - 12:44:34 | 200 |     422.574µs |   192.168.1.219 | GET      "/assets/close.svg"
[GIN] 2023/11/08 - 12:44:34 | 200 |     325.401µs |   192.168.1.219 | GET      "/assets/trash.svg"
[GIN] 2023/11/08 - 12:44:34 | 200 |    2.519234ms |   192.168.1.219 | GET      "/assets/img/favicons/favicon.ico"
[GIN] 2023/11/08 - 12:44:35 | 400 |    2.964904ms |   192.168.1.219 | POST     "/api/get/wrapperr-version"
[GIN] 2023/11/08 - 12:44:46 | 200 |     204.885µs |   192.168.1.219 | GET      "/admin"
[GIN] 2023/11/08 - 12:44:46 | 200 |    1.496444ms |   192.168.1.219 | GET      "/assets/css/admin.css"
[GIN] 2023/11/08 - 12:44:46 | 200 |     404.004µs |   192.168.1.219 | GET      "/js/admin.js"
[GIN] 2023/11/08 - 12:44:46 | 200 |     680.065µs |   192.168.1.219 | GET      "/js/adminFunctions.js"
[GIN] 2023/11/08 - 12:44:46 | 400 |      565.11µs |   192.168.1.219 | POST     "/api/get/wrapperr-version"
2023/11/08 12:44:46 Retrieved Wrapperr admin state. - Origin: 192.168.1.219
[GIN] 2023/11/08 - 12:44:46 | 200 |     448.001µs |   192.168.1.219 | POST     "/api/get/admin-state"
2023/11/08 12:44:49 Created and retrieved admin login JWT Token. - Origin: 192.168.1.219
[GIN] 2023/11/08 - 12:44:49 | 200 |    2.559846ms |   192.168.1.219 | POST     "/api/login/admin"
[GIN] 2023/11/08 - 12:44:49 | 200 |      39.889µs |   192.168.1.219 | GET      "/admin"
[GIN] 2023/11/08 - 12:44:49 | 400 |     568.217µs |   192.168.1.219 | POST     "/api/get/wrapperr-version"
2023/11/08 12:44:49 Retrieved Wrapperr admin state. - Origin: 192.168.1.219
[GIN] 2023/11/08 - 12:44:49 | 200 |      97.983µs |   192.168.1.219 | POST     "/api/get/admin-state"

**Comment by Jomack16** [comment]
above is the output from docker logs -f plexwrapper

I am unable to login to the admin panel after creating the admin user.
I am using the firefox browser, if that makes a difference.

**Comment by cyberweasel** [comment]
Having a similar problem.  It's been working fine but today tried to open and it said unconfigured.  I tried to admin username and pass and then it goes to admin login but entering that info just goes back to the same page.  I am seeing 400 errors in the logs too.  I see this line....New admin account created. Server is now claimed. but then I can't login.  I tried pulling the docker image again.

**Comment by aunefyren** [comment]
v3.2.0 had some major front end changes, particularly for JS. 

@cyberweasel or @Jomack16, can one of you confirm visiting the site in private mode (which ignores the cached version) doesn't fix this issue?

**Comment by Jomack16** [comment]
hi, 
just tried in a private window. same issue unfortunately.
attached .HAR of that attempt.
[private window HAR.zip](https://github.com/aunefyren/wrapperr/files/13310029/private.window.HAR.zip)

**Comment by Animosity022** [comment]
Hmm, same issue here as I created a new docker, tried a private window, different passwords and whatnot and won't let me login either.

**Comment by AlexanderEpolite** [comment]
Doing this for me as well, on the newest GitHub release zip and by cloning the repo directly.

**Comment by AlFontal** [comment]
I am having the same issue with wrapperr running on docker on a Raspberry Pi 4B.

Tried accessing the site from an M1 Macbook Pro both in normal and private browsing mode from Firefox, Chrome and Safari.

I was able to create the admin account but whenever I login it just gets stuck on the login panel, with the logs showing:
```
2023/11/12 09:28:52 Created and retrieved admin login JWT Token. - Origin: 192.168.1.135 

[GIN] 2023/11/12 - 09:28:52 | 200 |    6.080638ms |   192.168.1.135 | POST     "/api/login/admin"

[GIN] 2023/11/12 - 09:28:52 | 200 |      237.83µs |   192.168.1.135 | GET      "/admin"

[GIN] 2023/11/12 - 09:28:52 | 304 |     151.257µs |   192.168.1.135 | GET      "/js/functions.js"

[GIN] 2023/11/12 - 09:28:52 | 304 |     158.128µs |   192.168.1.135 | GET      "/js/adminFunctions.js"

[GIN] 2023/11/12 - 09:28:52 | 304 |     121.202µs |   192.168.1.135 | GET      "/js/admin.js"

[GIN] 2023/11/12 - 09:28:52 | 304 |     930.487µs |   192.168.1.135 | GET      "/assets/css/admin.css"

[GIN] 2023/11/12 - 09:28:52 | 400 |     1.68468ms |   192.168.1.135 | POST     "/api/get/wrapperr-version"
```

**Comment by ollebro** [comment]
I have the same issues as AlFontal, same logs as well.

**Comment by aunefyren** [comment]
I am having issues replicating this issue, but might have figured out the cause. Can someone who is experiencing this try out the newest `main` branch by compiling it with:
```
$ go build .
```

or the `beta` Docker image: `ghcr.io/aunefyren/wrapperr:beta`.

Make sure to entire refresh the page/try an incognito window to make sure your browser is not caching any old data.

**Comment by Animosity022** [comment]
I pulled the beta tag from the docker, stopped the instance, removed any old data, started a private window and tested that route, but got the same error.

Logs are very minimal.

```
Log file set.
2023/11/13 06:56:43 Running version v3.2.1.
2023/11/13 06:56:43 Starting Wrapperr on port 8282.
2023/11/13 06:56:43 Starting using HTTP.


[GIN] 2023/11/13 - 06:56:58 | 200 |    1.388286ms |    192.168.1.98 | GET      "/admin"
2023/11/13 06:56:58 Admin config file does not exist. Creating.
2023/11/13 06:56:58 Retrieved Wrapperr version. - Origin: 192.168.1.98
[GIN] 2023/11/13 - 06:56:58 | 200 |    2.587574ms |    192.168.1.98 | POST     "/api/get/wrapperr-version"
2023/11/13 06:56:58 Retrieved Wrapperr admin state. - Origin: 192.168.1.98
[GIN] 2023/11/13 - 06:56:58 | 200 |     197.286µs |    192.168.1.98 | POST     "/api/get/admin-state"
2023/11/13 06:57:08 New admin account created. Server is now claimed.
[GIN] 2023/11/13 - 06:57:08 | 201 |    3.269695ms |    192.168.1.98 | POST     "/api/create/admin"
[GIN] 2023/11/13 - 06:57:08 | 200 |     136.343µs |    192.168.1.98 | GET      "/admin"
2023/11/13 06:57:08 Retrieved Wrapperr version. - Origin: 192.168.1.98
[GIN] 2023/11/13 - 06:57:08 | 200 |     3.53591ms |    192.168.1.98 | POST     "/api/get/wrapperr-version"
2023/11/13 06:57:08 Retrieved Wrapperr admin state. - Origin: 192.168.1.98
[GIN] 2023/11/13 - 06:57:08 | 200 |     248.931µs |    192.168.1.98 | POST     "/api/get/admin-state"
2023/11/13 06:57:13 Admin login failed. Incorrect admin username or password. - Origin: 192.168.1.98
[GIN] 2023/11/13 - 06:57:13 | 401 |    4.001438ms |    192.168.1.98 | POST     "/api/login/admin"
```

**Comment by aunefyren** [comment]
> 2023/11/13 06:57:13 Admin login failed. Incorrect admin username or password. - Origin: 192.168.1.98

Could this be something else? I see you got a 401 response and a complaint about bad credentials. Could you have typed the credentials wrong, or perhaps something is broken regarding the admin creation?

**Comment by Animosity022** [comment]
I'd say admin creation and whatever it is setting the password as. I tried 3 times with private browsers and I'm awake enough to not type the wrong password numerous times.

**Comment by Jomack16** [comment]
I just pulled the beta as requested. The mainpage has changed for me. It now has a snowing background and a gift on the sign in section.
It says 'wrapperr is not configured' in red.
I click on Admin at the bottom of the screen and create an admin user account.
Then when I try to login to the admin page, it fails again.
This time there is no 400 error
HAR file and wrapper.log attached.
[HAR and Log.zip](https://github.com/aunefyren/wrapperr/files/13342567/HAR.and.Log.zip)

**Comment by aunefyren** [comment]
Hey, @Jomack16, your log file says:

> 2023/11/07 10:22:45 Running version v3.2.0.

Please try v3.2.1 as I mentioned above, like @AlexanderEpolite.

**Comment by Jomack16** [comment]
I was not using localhost or https in my testing, so that makes sense.
Using the :beta tag in my docker command now.
It is working! I am able to login to the admin page after creating a user.

**Comment by aunefyren** [comment]
Awesome @Jomack16, I will push these changes as soon as possible!

Thank you to everyone who helped resolve this.

**Comment by Jeanball** [comment]
Hi,

I have the same issue of @Jomack16 and when I tried with beta tag, everything is up and running! I was using ZeroTier to reach my wrapperr instance so yeah it's explain why it's not working with strict mode enable. Thanks for your work!

**Comment by aunefyren** [comment]
Fixed in v3.2.1.
